src/util.rs

```diff
@@ -55,7 +55,12 @@ impl Fairing for AppHeaders {
         res.set_raw_header("Referrer-Policy", "same-origin");
         res.set_raw_header("X-Content-Type-Options", "nosniff");
         res.set_raw_header("X-Robots-Tag", "noindex, nofollow");
-        res.set_raw_header("Cross-Origin-Resource-Policy", "same-origin");
+
+        if let Some(path_stripped) = req_uri_path.strip_prefix(&*CONFIG.domain_path()) {
+            if !(path_stripped.starts_with("/icons/") || path_stripped.starts_with("/vw_static/")) {
+                res.set_raw_header("Cross-Origin-Resource-Policy", "same-origin");
+            }
+        }
```

Comment:
I think I'd also prefer @BlackDex suggestion. I.e, something like:
```suggestion
        if !res.is_png() {
            res.set_raw_header("Cross-Origin-Resource-Policy", "same-origin");
        }
```

Reply:
What about non-PNG images?
In my opinion, it's too much to not apply Cross-Origin-Resource-Policy to all images.

Reply:
```suggestion
        if !res.headers().get_one("Content-Type").is_some_and(|v| v.starts_with("image")) {
            res.set_raw_header("Cross-Origin-Resource-Policy", "same-origin");
        }
```

Reply:
I don't think that we can have it both ways without complicating the this further. I think allowing only `PNG` should be enough. (If you don't agree, I've also posted an suggestion above https://github.com/dani-garcia/vaultwarden/pull/5469#discussion_r1930244657 which would only allow the bare minimum.)

Reply:
In the future, we may use WebP or AVIF, so [this](https://github.com/dani-garcia/vaultwarden/pull/5469#discussion_r1930318145) method is a good choice if you do not want to assign Cross-Origin-Resource-Policy to all images.

Reply:
That isn't possible. The content type is based upon the image type by it's headers. And thus not only `PNG` types are returned.

https://github.com/dani-garcia/vaultwarden/blob/2903a3a13ad06a09805e8842511b4b715a686d8e/src/api/icons.rs#L582-L592

Reply:
And:
https://github.com/dani-garcia/vaultwarden/blob/2903a3a13ad06a09805e8842511b4b715a686d8e/src/api/icons.rs#L113-L119